src/fi_tostr.c

```diff
@@ -537,6 +537,102 @@ static void ofi_tostr_fabric_attr(char *buf, const struct fi_fabric_attr *attr,
 		FI_MAJOR(attr->api_version), FI_MINOR(attr->api_version));
 }
 
+static void ofi_tostr_device_attr(char *buf, const struct fi_device_attr *attr)
+{
+	const char *prefix = TAB TAB;
+	ofi_strcatf(buf, "%sfi_device_attr:\n", prefix);
+
+	prefix = TAB TAB TAB;
+	ofi_strcatf(buf, "%sname: %s\n", prefix, attr->name);
+	ofi_strcatf(buf, "%sdevice_id: %s\n", prefix, attr->device_id);
+	ofi_strcatf(buf, "%sdevice_version: %s\n", prefix, attr->device_version);
+	ofi_strcatf(buf, "%svendor_id: %s\n", prefix, attr->vendor_id);
+	ofi_strcatf(buf, "%sdriver: %s\n", prefix, attr->driver);
+	ofi_strcatf(buf, "%sfirmware: %s\n", prefix, attr->firmware);
+}
+
+static void ofi_tostr_pci_attr(char *buf, const struct fi_pci_attr *attr)
+{
+	const char *prefix = TAB TAB TAB;
+	ofi_strcatf(buf, "%sfi_pci_attr:\n", prefix);
+
+	prefix = TAB TAB TAB TAB;
+	ofi_strcatf(buf, "%sdomain_id: %u\n", prefix, attr->domain_id);
+	ofi_strcatf(buf, "%sbus_id: %u\n", prefix, attr->bus_id);
+	ofi_strcatf(buf, "%sdevice_id: %u\n", prefix, attr->device_id);
+	ofi_strcatf(buf, "%sfunction_id: %u\n", prefix, attr->function_id);
+}
+
+static void ofi_tostr_bus_type(char *buf, int type)
+{
+	switch (type) {
+	CASEENUMSTR(FI_BUS_UNKNOWN);
+	CASEENUMSTR(FI_BUS_PCI);
```

Comment:
Shouldn't FI_BUS_PCI output "PCI" or similar?

Reply:
CASEENUMSTR will output the the name as a string, so this will be "FI_BUS_PCI".

Reply:
That was me not reading carefully.  I think I even wrote that code... duh!